src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,87 @@
-import React from 'react';
+import cl from 'classnames';
+import React, { FormEvent, useState } from 'react';
+import { addComment } from '../api/api';
+import { pause } from '../utils/methods';
+import { Comment } from '../types/Comment';
+
+interface Props {
+  postId: number;
+  onAddComment: (v: Comment) => void;
+}
+
+export const NewCommentForm: React.FC<Props> = ({ postId, onAddComment }) => {
+  const [nameError, setNameError] = useState(false);
+  const [emailError, setEmailError] = useState(false);
+  const [bodyError, setBodyError] = useState(false);
+
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [body, setBody] = useState('');
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleAddChange = async () => {
+    setNameError(!name);
+    setEmailError(!email);
+    setBodyError(!body);
+
+    if (!name || !email || !body) {
+      return;
```

Comment:
Consider providing user feedback when validation fails, such as displaying a message or highlighting the fields.